core/server/data/migrations/versions/4.8/04-migrate-show-newsletter-header-setting.js

```diff
@@ -0,0 +1,98 @@
+//
+// the default for both of the new settings is true so there's no need to migrate
+// anything if previously showing the header
+
+module.exports = createTransactionalMigration(
+    async function up(connection) {
+        const oldSetting = await connection('settings')
+            .where('key', 'newsletter_show_header')
+            .select('value')
+            .first();
+
+        if (!oldSetting) {
+            logging.info('Skipping migration of newsletter_show_header setting. Does not exist');
+            return;
+        }
+
+        if (oldSetting.value === 'false') {
+            logging.info('Migrating newsletter_show_header setting. Disabling title and logo');
+            await connection('settings')
+                .whereIn('key', ['newsletter_show_header_title', 'newsletter_show_header_icon'])
+                .update({value: 'false'});
+        } else {
+            logging.info('Skipping migration of newsletter_show_header setting. Matches new defaults');
+        }
+
+        logging.info('Deleting newsletter_show_header setting');
+        await connection('settings')
+            .where('key', 'newsletter_show_header')
+            .del();
+    },
+
+    async function down(connection) {
+        logging.info('Adding newsletter_show_header setting');
+
+        const oldSetting = await connection('settings')
+            .where('key', '=', 'newsletter_show_header')
+            .select('value')
+            .first();
+
+        const showTitle = await connection('settings')
+            .where('key', '=', 'newsletter_show_header_title')
+            .select(['value', 'created_by'])
+            .first();
+
+        const showIcon = await connection('settings')
+            .where('key', '=', 'newsletter_show_header_icon')
+            .select(['value', 'created_by'])
+            .first();
+
+        if (oldSetting) {
+            logging.warn('Skipping rollback of newsletter_show_header setting. Already exists');
+        } else {
+            const [{id: ownerId} = {id: 1}] = await connection('users')
+                .select('users.id')
+                .innerJoin(
+                    'roles_users',
+                    'users.id',
+                    'roles_users.user_id'
+                )
+                .where(
+                    'roles_users.role_id',
+                    connection('roles').select('id').where('name', 'Owner')
+                );
+
+            let showHeader = true; // default
+
+            if (showTitle && showIcon) {
+                showHeader = showTitle.value === 'false' && showIcon.value === 'false' ? false : true;
+            }
+
+            await connection('settings')
+                .insert({
+                    id: ObjectId().toHexString(),
+                    key: 'newsletter_show_header',
+                    group: 'newsletter',
+                    type: 'boolean',
+                    value: showHeader,
+                    created_at: moment().toDate(),
```

Comment:
This inserts the wrong thing into the DB for SQLite for me:
* inserted: `1624267454065`
* other values: `2021-06-17 16:20:03`

Reply:
Ugh, OK. I'll switch to the `knex.raw('CURRENT_TIMESTAMP')` pattern although that also has the non-UTC issue.

Reply:
Updated